.github/workflows/builder.yml

```diff
@@ -54,18 +51,37 @@ jobs:
   #                       Build the builder                         #
   #                                                                 #
   ###################################################################
+  detect-env:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # Needed to detect the current reusable repository and ref.
+      contents: read
+    outputs:
+      builder_repo: ${{ steps.detect.outputs.builder_repo }}
+      builder_ref: ${{ steps.detect.outputs.builder_ref }}
+    steps:
+      - name: Detect the builder ref
+        id: detect
+        shell: bash
+        run: | 
+          export OIDC_TOKEN=$(curl -sS -H "Authorization: Bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN" "$ACTIONS_ID_TOKEN_REQUEST_URL" | jq -r '.value')
```

Comment:
can you add a specific audience, like in https://github.com/slsa-framework/slsa-github-generator-go/blob/main/pkg/provenance.go#L291? Maybe you can also update the one we use in this file to something more specific, e.g., `slsa-framework/slsa-github-generator-go/_some_usecase`. I want to be sure a leak of this token does not allow forging provenance signatures.

Reply:
I made it the fully qualified URL here ...
https://github.com/slsa-framework/slsa-github-generator-go

Reply:
even better :-)